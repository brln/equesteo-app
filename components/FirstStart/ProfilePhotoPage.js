import ImagePicker from 'react-native-image-crop-picker'

import React, { PureComponent } from 'react'
import {
  Dimensions,
  StyleSheet,
  Text,
  TouchableOpacity,
  View
} from 'react-native'

import BuildImage from '../Images/BuildImage'
import { brand } from '../../colors'
import { logRender } from '../../helpers'
import Button from '../Button'
import MedImage from '../Images/MedImage'

const { width } = Dimensions.get('window')

export default class ProfilePhotoPage extends PureComponent {
  constructor () {
    super()
    this.uploadProfile = this.uploadProfile.bind(this)
  }

  uploadProfile () {
    ImagePicker.openPicker({
      width: 1080,
      height: 1080,
      cropping: true
    }).then(image => {
      this.props.uploadProfilePhoto(image.path)
    }).catch(() => {})
  }

  render() {
    logRender('ProfilePhotoPage')
    let button = (
      <Button
        color={brand}
        text={"Yes!"}
        onPress={this.uploadProfile}
      />
    )
    let skip = (
      <TouchableOpacity onPress={this.props.nextPage} style={{paddingTop: 10}}>
        <Text style={{fontSize: 10, textAlign: 'center', textDecorationLine: 'underline'}}>No, thanks.</Text>
      </TouchableOpacity>
    )
    let profilePhoto = (
      <TouchableOpacity onPress={this.uploadProfile} >
        <BuildImage
          source={require('../../img/emptyProfile.png')}
          style={{width: '100%', height: '100%'}}
        />
      </TouchableOpacity>
    )

    if (this.props.user.get('profilePhotoID')) {
      button = (
        <Button
          color={brand}
          text={"You look great! Continue"}
          onPress={this.props.nextPage}
        />
      )
      skip = null
      profilePhoto = (
        <MedImage
          source={{uri: this.props.userPhotos.getIn([this.props.user.get('profilePhotoID'), 'uri'])}}
          style={{width: '100%', height: '100%'}}
          onError={e => logError("Can't load FirstProfilePhoto image")}
        />
      )
    }

    return (
      <View style={{flex: 1}}>
        <View style={{paddingLeft: 20, paddingRight: 20}}>
          <Text style={{
            fontSize: 30,
            fontFamily: 'Montserrat-Regular',
            color: 'black',
            textAlign: 'center'
          }}>
            Upload a profile photo?
          </Text>
        </View>
        <View style={{flex: 1, alignItems: 'center'}}>
          <View style={{paddingTop: 30, paddingLeft: 10, paddingRight: 10, width: width * .5, height: width * .5}}>
            <View style={{borderColor: brand, borderWidth: 5, backgroundColor: 'white'}}>
              { profilePhoto }
            </View>
          </View>
          <View style={{paddingTop: 20}}>
            { button }
            { skip }
          </View>
        </View>
      </View>
    )
  }
}

const styles = StyleSheet.create({
});
