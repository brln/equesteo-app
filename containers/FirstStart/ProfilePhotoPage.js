import ImagePicker from 'react-native-image-crop-picker'

import React, { PureComponent } from 'react'
import {
  Alert,
  Dimensions,
  Text,
  TouchableOpacity,
  View
} from 'react-native'
import { connect } from 'react-redux'

import BuildImage from '../../components/Images/BuildImage'
import { brand } from '../../colors'
import { logInfo, logRender, logError, generateUUID, unixTimeNow } from '../../helpers'
import Button from '../../components/Button'
import MedImage from '../../components/Images/MedImage'
import {
  createUserPhoto,
  userUpdated,
} from '../../actions/standard'
import functional from '../../actions/functional'
import EqNavigation from "../../services/EqNavigation"
import { FIRST_HORSE_PAGE } from "../../screens/consts/firstStart"
import Wrapper from '../../components/FirstStart/Wrapper'

const { width } = Dimensions.get('window')

class ProfilePhotoPage extends PureComponent {
  static options() {
    return {
      topBar: {
        background: {
          color: 'transparent',
        },
        elevation: 0,
        backButton: {
          color: brand,
        },
      },
      layout: {
        orientation: ['portrait']
      }
    }
  }

  constructor () {
    super()
    this.nextPage = this.nextPage.bind(this)
    this.uploadProfile = this.uploadProfile.bind(this)
  }

  uploadProfile () {
    ImagePicker.openPicker({
      width: 1080,
      height: 1080,
      cropping: true
    }).then(image => {
      this.uploadProfilePhoto(image.path)
    }).catch(e => {
      if (e.code && e.code === 'E_PERMISSION_MISSING') {
        Alert.alert('Denied', 'You denied permission to access photos. Please enable via permissions settings for Equesteo.')
      } else {
        logError(e, 'Containers.FirstStart.uploadProfile')
      }
    })
  }

  uploadProfilePhoto (location) {
    let photoID = generateUUID()
    let userID = this.props.user.get('_id')
    this.props.dispatch(createUserPhoto(
      userID,
      {
        _id: photoID,
        timestamp: unixTimeNow(),
        uri: location
      }
    ))
    this.props.dispatch(userUpdated(this.props.user.set('profilePhotoID', photoID)))
    this.props.dispatch(functional.persistUserWithPhoto(userID, photoID, false))
  }

  nextPage () {

    EqNavigation.push(this.props.componentId, {
      component: {
        name: FIRST_HORSE_PAGE,
      }
    }).catch(() => {})
  }

  render() {
    logRender('ProfilePhotoPage')
    let button = (
      <Button
        color={brand}
        text={"Yes!"}
        onPress={this.uploadProfile}
      />
    )
    let skip = (
      <TouchableOpacity onPress={this.nextPage} style={{paddingTop: 10}}>
        <Text style={{fontSize: 10, textAlign: 'center', textDecorationLine: 'underline'}}>No, thanks.</Text>
      </TouchableOpacity>
    )
    let profilePhoto = (
      <TouchableOpacity onPress={this.uploadProfile} >
        <BuildImage
          source={require('../../img/emptyProfile.png')}
          style={{width: '100%', height: '100%'}}
        />
      </TouchableOpacity>
    )

    if (this.props.user.get('profilePhotoID')) {
      button = (
        <Button
          color={brand}
          text={"You look great! Continue"}
          onPress={this.nextPage}
        />
      )
      skip = null
      profilePhoto = (
        <MedImage
          source={{uri: this.props.userPhotos.getIn([this.props.user.get('profilePhotoID'), 'uri'])}}
          style={{width: '100%', height: '100%'}}
          onError={e => logInfo("Can't load FirstProfilePhoto image")}
        />
      )
    }

    return (
      <Wrapper>
        <View style={{flex: 1}}>
          <View style={{paddingLeft: 20, paddingRight: 20}}>
            <Text style={{
              fontSize: 30,
              fontFamily: 'Montserrat-Regular',
              color: 'black',
              textAlign: 'center'
            }}>
              Upload a profile photo?
            </Text>
          </View>
          <View style={{flex: 1, alignItems: 'center'}}>
            <View style={{paddingTop: 30, paddingLeft: 10, paddingRight: 10, width: width * .5, height: width * .5}}>
              <View style={{borderColor: brand, borderWidth: 5, backgroundColor: 'white'}}>
                { profilePhoto }
              </View>
            </View>
            <View style={{paddingTop: 20}}>
              { button }
              { skip }
            </View>
          </View>
        </View>
      </Wrapper>
    )
  }
}

function mapStateToProps (state, passedProps) {
  const localState = state.get('localState')
  const pouchState = state.get('pouchRecords')
  const horseID = localState.getIn(['firstStartHorseID', 'horseID'])
  return {
    horse: pouchState.getIn(['horses', horseID]),
    user: pouchState.getIn(['users', localState.get('userID')]),
    userID: localState.get('userID'),
    userPhotos: pouchState.get('userPhotos')
  }
}

export default connect(mapStateToProps)(ProfilePhotoPage)
