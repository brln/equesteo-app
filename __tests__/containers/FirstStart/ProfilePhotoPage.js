import configureStore from 'redux-mock-store'
import React from "react"
import {fromJS, Map} from "immutable"
import renderer from "react-test-renderer"
import { Provider } from 'react-redux'
import thunk from "redux-thunk"

import ProfilePhotoPage from '../../../containers/FirstStart/ProfilePhotoPage'


const mockStore = configureStore([thunk])

test('render', () => {
  const store = mockStore(fromJS({
    localState: {
      userID: 'userID1'
    },
    pouchRecords: {
      users: {
        userID1: {
          _id: 'userID1'
        }
      }
    },
  }))

  const tree = renderer.create(
    <Provider store={store}>
      <ProfilePhotoPage />
    </Provider>)
  expect(tree.toJSON()).toMatchSnapshot()
})
