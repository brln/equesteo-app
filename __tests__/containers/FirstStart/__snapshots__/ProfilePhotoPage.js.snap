// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render 1`] = `
<View>
  <View
    style={
      Object {
        "height": 1278,
        "position": "absolute",
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "white",
          "flex": 3,
          "justifyContent": "flex-end",
        }
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../img/firstStart.jpg",
          }
        }
        style={
          Object {
            "height": 750,
            "resizeMode": "cover",
            "width": 750,
          }
        }
      />
    </View>
  </View>
  <RCTScrollView
    automaticallyAdjustContentInsets={false}
    contentInset={
      Object {
        "bottom": 0,
      }
    }
    enableAutomaticScroll={true}
    enableOnAndroid={false}
    enableResetScrollToCoords={true}
    extraHeight={75}
    extraScrollHeight={0}
    getScrollResponder={[Function]}
    handleOnScroll={[Function]}
    keyboardDismissMode="interactive"
    keyboardOpeningTime={250}
    keyboardSpace={0}
    onScroll={[Function]}
    resetKeyboardSpace={[Function]}
    scrollEventThrottle={1}
    scrollForExtraHeightOnAndroid={[Function]}
    scrollIntoView={[Function]}
    scrollToEnd={[Function]}
    scrollToFocusedInput={[Function]}
    scrollToPosition={[Function]}
    showsVerticalScrollIndicator={true}
    style={
      Object {
        "backgroundColor": "transparent",
      }
    }
    update={[Function]}
    viewIsInsideTabBar={false}
  >
    <View>
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "white",
            "flexDirection": "column",
            "paddingTop": 20,
          }
        }
      >
        <Image
          source={1}
          style={
            Object {
              "alignItems": "center",
              "height": 60,
              "width": 60,
            }
          }
        />
      </View>
      <View
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <View
          style={
            Object {
              "paddingLeft": 20,
              "paddingRight": 20,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "black",
                "fontFamily": "Montserrat-Regular",
                "fontSize": 30,
                "textAlign": "center",
              }
            }
          >
            Upload a profile photo?
          </Text>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
            }
          }
        >
          <View
            style={
              Object {
                "height": 375,
                "paddingLeft": 10,
                "paddingRight": 10,
                "paddingTop": 30,
                "width": 375,
              }
            }
          >
            <View
              style={
                Object {
                  "backgroundColor": "white",
                  "borderColor": "#5bc0eb",
                  "borderWidth": 5,
                }
              }
            >
              <View
                accessible={true}
                clickable={true}
                isTVSelectable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
              >
                <Image
                  source={1}
                  style={
                    Object {
                      "height": "100%",
                      "width": "100%",
                    }
                  }
                />
              </View>
            </View>
          </View>
          <View
            style={
              Object {
                "paddingTop": 20,
              }
            }
          >
            <View
              accessible={true}
              clickable={true}
              isTVSelectable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignSelf": "center",
                  "backgroundColor": "#5bc0eb",
                  "borderColor": undefined,
                  "borderRadius": 10,
                  "borderWidth": null,
                  "marginTop": 10,
                  "opacity": 1,
                  "paddingBottom": 10,
                  "paddingTop": 10,
                  "width": undefined,
                }
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#fff",
                      "paddingLeft": 20,
                      "paddingRight": 20,
                      "textAlign": "center",
                    },
                    undefined,
                  ]
                }
              >
                Yes!
              </Text>
            </View>
            <View
              accessible={true}
              clickable={true}
              isTVSelectable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                  "paddingTop": 10,
                }
              }
            >
              <Text
                style={
                  Object {
                    "fontSize": 10,
                    "textAlign": "center",
                    "textDecorationLine": "underline",
                  }
                }
              >
                No, thanks.
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
